src/main/java/uk/ac/ebi/ega/egafuse/service/EgaRetryService.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.service;
+
+import static uk.ac.ebi.ega.egafuse.config.EgaFuseApplicationConfig.CHUNK_SIZE;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.google.common.io.CountingInputStream;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import uk.ac.ebi.ega.egafuse.exception.ClientProtocolException;
+import uk.ac.ebi.ega.egafuse.model.CacheKey;
+
+public class EgaRetryService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EgaRetryService.class);
+    private OkHttpClient okHttpClient;
+    private String apiURL;
+    private Token token;
+
+    public EgaRetryService(OkHttpClient okHttpClient, String apiURL, Token token) {
+        this.okHttpClient = okHttpClient;
+        this.apiURL = apiURL;
+        this.token = token;
+    }
+
+    @Retryable(value = {IOException.class, ClientProtocolException.class}, maxAttemptsExpression = "${connection.maxAttempts}", 
+            backoff = @Backoff(delayExpression = "${connection.backoff}"))

```

Comment:
You have put the backoff what is the maximum number of retries?

Reply:
connection.backoff is the maximum number of retries whose value is 6 in properties file